### PR TITLE
Updates with reviews component

### PR DESCRIPTION
**Pull request by mhhutton**

### CODE REVIEW

File: client/src/components/reviews/MetaData/ProductReviewChart.jsx [hunk 1]
```diff
@@ -84,109 +92,56 @@ function valuetext(value) {
...
+      let charNumVal =  revsMetaData.characteristics[key]['value'];
+      let charArr = [key, Math.round(Number(charNumVal)), getCharLowHighVals(key)];
+      resultArr.unshift(charArr);
+    })
+    setCharValPairs(resultArr);
+  }
+
+  useEffect( () => {
+    if (revsMetaData) {
+      if (revsMetaData.characteristics) {
```
Comment (Pix3lAssassin) [inline client/src/components/reviews/MetaData/ProductReviewChart.jsx:111]: Could be done on one line using an &

File: client/src/components/reviews/MetaData/ProductReviewChart.jsx [hunk 2]
```diff
@@ -84,109 +92,56 @@ function valuetext(value) {
...
         aria-labelledby="discrete-slider"
         valueLabelDisplay="auto"
         step={1}
-        marks
-        min={0}
+        marks={item[2]}
+        min={1}
         max={5}
         disabled
         />
```
Comment (Pix3lAssassin) [inline client/src/components/reviews/MetaData/ProductReviewChart.jsx:142]: should be indented

File: client/src/components/reviews/MetaData/ReviewPercentageAndStars.jsx [hunk 3]
```diff
@@ -1,9 +1,16 @@
-import React from 'react';
+import React, {useContext, useState, useEffect, useStateWithCallback} from 'react';
```
Comment (Pix3lAssassin) [inline client/src/components/reviews/MetaData/ReviewPercentageAndStars.jsx:1]: Remove unused useStateWithCallback

File: client/src/components/reviews/RatingsAndReviews.jsx [hunk 4]
```diff
@@ -30,20 +32,71 @@ const RatingsAndReviews = () => {
...
         }
       })
       .then((response) => {
+        //let newReviews = [...reviews, ...response.data.results]
+        //let newReviews = response.data.results
+        // setReviews(newReviews);
         setReviews(response.data);
+        //setReviewResults(newReviews);
       })
+      // .then((response) => {
```
Comment (ThinkCreatively) [inline client/src/components/reviews/RatingsAndReviews.jsx:47]: If youre not using both of these then blocks maybe get rid of one?

File: client/src/components/reviews/RatingsAndReviews.jsx [hunk 5]
```diff
@@ -30,20 +32,71 @@ const RatingsAndReviews = () => {
...
+        }
+      })
+      .then((response) => {
+        if(response.data) {
+          if (response.data.results) {
+            if (response.data.results.length < 1) {
+              setPageState(null);
+            } else {
+              let newReviews = response.data.results
+              //let newReviewsResults = [...reviewResults, ...newReviews]
```
Comment (ThinkCreatively) [inline client/src/components/reviews/RatingsAndReviews.jsx:79]: Double new Reviews let variable

File: client/src/components/reviews/RatingsAndReviews.jsx [hunk 6]
```diff
@@ -30,20 +32,71 @@ const RatingsAndReviews = () => {
...
+          }
+        }
+      })
+      .catch( (err) => {
+        console.log(err);
+        res.send(err);
+      })
+    //setReviewResults(reviews.results);
+    console.log('response results in getAllResults', reviewResults);
+    // var num = pageState;
```
Comment (ThinkCreatively) [inline client/src/components/reviews/RatingsAndReviews.jsx:95]: Code Cleanup commented lines for master

**Review by Pix3lAssassin — APPROVED** [review]

File: client/src/components/reviews/helperFunctions.jsx [hunk 7]
```diff
@@ -13,4 +13,125 @@ export function getTotalReviews(revData) {
...
+  let total = 0;
+  //This for loop adds all values in array in order to use resulting total as denominator for calculating percentages.
+  for (var i = 0; i < starValsArr.length; i += 1) {
+      total += starValsArr[i];
+  }
+
+  for (var i = 0; i < starValsArr.length; i += 1) {
+    resultArr.push(starValsArr[i] / total * 100);
+  }
+  return resultArr;
```
Comment (ThinkCreatively) [inline client/src/components/reviews/helperFunctions.jsx:136]: Add empty line after last line of code so linter isn't upset, I dont know the reason why but I think its best practcies

**Pull request closed without merging**